src/Parser.php

```diff
@@ -391,8 +351,22 @@ protected function parseScenario()
      */
     protected function parseOutline()
     {
-        $token = $this->expectTokenType('Outline');
+        $node = $this->parseScenarioOrOutlineBody($this->expectTokenType('Outline'));
+
+        if (!($node instanceof OutlineNode && $node->hasExamples())) {
+            throw new ParserException(sprintf(
+                'Outline should have examples table, but got none for outline "%s" on line: %d%s',
+                $node->getTitle(),
+                $node->getLine(),
+                $this->file ? ' in file: ' . $this->file : ''
+            ));
+        }
```

Comment:
NB cucumber/gherkin doesn't throw in this case (see #153) - I have kept it for now but plan to look at that separately.

Reply:
I think solving #153 fully is just a matter of removing that exception, by _fully_ making them synonyms.

Reply:
I agree, I just mean it's a separate PR/change. It relates to other cases where we currently throw on partially-complete feature files that gherkin accepts and I think we should consider that in the round.

Reply:
well, to me, accepting outlines without examples (returning them as a ScenarioNode) is part of making `Scenario` and `Scenario Outline` be synonyms. Right now, you are making `Scenario` a synonym of `Scenario Outline` but `Scenario Outline` is not a synonym of `Scenario`, which is quite weird. Synonyms are meant to be bidirectional relations to me.

Reply:
I would say that turning a parsing exception into a supported case won't break CI of anyone: if they have files matching that case, it will potentially _fix_ their CI. Turning invalid files into valid ones is generally not considered a BC break (otherwise, it would be impossible to add any new syntax)
And this is also exactly the kind of changes you are doing for files using `Scenario` that have an `Examples` table today (which are also a ParserException).

But anyway, I won't block the merge on that if you decide to keep it split.

Reply:
Fair points and broadly I agree, I just think there's a philosophical difference between changing behaviour that is undefined, and changing behaviour that we have explicit tests for. Hence I prefer the discipline of considering the second type of changes separately. The next PR will be very soon :smile: